oxo.py

```diff
@@ -1,26 +1,52 @@
 class OxoBoard:
+
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.tiles_on_board = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+        self.PLAYER_CROSS = 1
+        self.PLAYER_NAUGHT = 2
+        self.amount_for_win = 3
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.tiles_on_board[x][y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.tiles_on_board[x][y] is 0:
+            self.tiles_on_board[x][y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for x in range(len(self.tiles_on_board)):
+            for y in range(len(self.tiles_on_board)):
+                if self.tiles_on_board[x][y] is 0:
+                    return False
+        return True
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        for player in range(1, 3):
+            diagonal_a = self.tiles_on_board[0][0] is player and self.tiles_on_board[1][1] is player\
```

Comment:
Any reason you are using `is` rather than `==` here? Either works but the latter is maybe more conventional